apps/opencs/view/render/terraintexturemode.hpp

```diff
@@ -102,14 +107,15 @@ namespace CSVRender
             bool allowLandTextureEditing(std::string textureFileName);
 
             std::string mCellId;
-            std::string mBrushTexture;
-            int mBrushSize;
-            int mBrushShape;
+            std::string mBrushTexture = "L0#0";
```

Comment:
Not sure which rules we have adopted for C+11 in this regard (that was probably while I was away). But IMO it is a bad idea to initialise variables this way, if the class also has a constructor. Use one or the other. Don't mix them.

Reply:
For terrainshapemode, I got a review from Elsid, who suggested moving initialization here as for some C-standard (maybe C+11 standard) , but of course it should be consistently the same way. This was adopted for terrainshapemode, so this is the reason I'm doing the same here, as I was forced to do minor changes here anyway.

Reply:
Suggesting to fix this so that I remove these from terraintexturemode.cpp, and keep initialization only at the header. This is a matter of standard, and I have nothing to argue why one is better than the other, so will do either way.

Reply:
I do. If you have it in the header, than any modification has the potential to trigger re-compiling of mulitple cpp files, while making a change to a cpp will only force re-compliation of a single cpp file.
We need to write something about that into our style guide, if we haven't already. IMO initialising members in the class definition should be limited to basic classes that are just collections of data (typically structs) and constants where feasible.

Reply:
I now removed the initializations from terraintexturemode header. Terrainshapemode is untouched, suggesting to change that to same style in another PR.

Reply:
Using in-class initializers is a part of C++ Core Guidelines: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c48-prefer-in-class-initializers-to-member-initializers-in-constructors-for-constant-initializers. Of course we can create our own different style guide rules but then we face [this](https://imgs.xkcd.com/comics/standards.png). When there is an existing standard covering some case better to use it rather than propose another standard. Also it's always possible to propose a change to a standard. Argument about compilation time should be based on some statistics (how often defaults change per commit). Because if all default initializations changes increase compilation time across all builds by insignificant percent then there is no much sense to have this argument. I think having less copy-paste and no forgotten default initalizations is a stronger argument.

Reply:
Admittedly the compile time argument is pretty weak. I only brought it up to show there is one, after unelsson wasn't able to find any.

I don't care hugely about other people's style guide, particularly in regards to c++11 which is still pretty new. Each project should define its own, but some conformity is certainly desirable. Overall the question of prefer constructor or prefer in class initialisation is again subjective. My main point here was that we shouldn't mix them in the same class. Because that is not subjective at all.

Reply:
I agree with this sentiment.

To be fair, we're on C++14 though switching to C++17 is possible, we're kneecapped by MacOS and Android not supporting nice things like std::filesystem (yet).